### ISSUE TITLE
[BUG] Save() with single Path fails to parse "received"

### ISSUE THREAD

**Issue by john08burke** [OpenSIPS/opensips#1809]
**OpenSIPS version you are running**
```
version: opensips 2.4.6 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC, FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16, MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
main.c compiled on 16:23:30 Jul 18 2019 with gcc 6.3.0
```

**Describe the bug**
When using single Path header from sbc to registrar, save("v") fails to parse path params and consequently received-uri, NAT branch flag are not saved in usrloc.

71c7d75f66c8f0dc4ce411fd97d2dc867b39ce8d seems to be the culprit, where double call to `parse_params()` without refilling params occurs if not using double-Path. https://github.com/OpenSIPS/opensips/blob/11d7dcc28d9a4ae9351e12c5a5d1f9a2da2d7279/lib/reg/path.c#L89-L109

**To Reproduce**
1. Send REGISTER with single Path header, inclusive of `received` param.
```
REGISTER sip:test.domain.net;transport=UDP SIP/2.0
...
...
Path: <sip:192.168.20.164;lr;received=sip:192.168.51.4:5062>
...
```

2. Check in-memory usrloc via ul_dump.
3. Contact from ul_dump will not contain "Received" and "Cflags" does not contain nat_bflag("...").

**Expected behavior**
Expecting both Received and Cflags to be populated correctly from ul_dump.

**Relevant System Logs**

**OS/environment information**
 - Operating System: Debian 9
 - OpenSIPS installation: git

**Comment by liviuchircu** [comment]
Hi @john08burke !  Can you cherry-pick 8070dac142fc6 into your `2.4` branch and see if it fixes the issue?  Thanks!

**Comment by liviuchircu** [comment]
Upon further inspection, this is a different issue than the one fixed by 8070dac, as most of my testing was done with `save("p1v")`, not `save("v")`.  Could you please pull 45dd0e560 in your `2.4` tree with `git cherry-pick -x 45dd0e560` then rebuild and see if the issue goes away?  Thanks!

**Comment by john08burke** [comment]
@liviuchircu, I cherry-picked 45dd0e5600506c140cc7bbc1485f6c567c43e27e and everything works as expected.  Thanks!
